src/load/azext_load/data_plane/load_test/custom.py

```diff
@@ -136,8 +146,33 @@ def create_test(
     upload_files_helper(
         client, test_id, yaml, test_plan, load_test_config_file, not custom_no_wait, evaluated_test_type
     )
-    response = client.get_test(test_id)
     logger.info("Upload files to test %s has completed", test_id)
+    if is_not_empty_dictionary(app_components):
+        # only get and patch the app components if its present in the yaml.
+        app_component_response = client.create_or_update_app_components(
+            test_id=test_id, body={"testId": test_id, "components": app_components}
+        )
+        logger.warning(
+            "Added app components for test ID: %s and response is %s", test_id, app_component_response
+        )
+    if is_not_empty_dictionary(server_metrics):
+        # only get and patch the app components if its present in the yaml.
+        server_metrics_existing = None
+        try:
+            server_metrics_existing = client.get_server_metrics_config(test_id)
+        except Exception:
+            server_metrics_existing = {"metrics": {}}
```

Comment:
Shouldnt do this for any exception since 503 can result in invalid merge?

Reply:
i dont think we need to retry these things, we can just do the patch call for the app-components, and moreover adding retries should have been at the client level ie. sdk so i dont think we need to add anything seperatelyt.